src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggingExtensions.cs

```diff
@@ -79,6 +82,25 @@ public static class OpenTelemetryLoggingExtensions
                     disposeProvider: false)));
 
         return builder;
+
+        // The warning here is about the fact that the OpenTelemetryLoggerOptions will be bound to configuration using ConfigurationBinder
+        // That uses reflection a lot - so if any of the properties on that class were complex types reflection would be used on them
+        // and nothing could guarantee its correctness.
+        // Since currently this class only contains primitive (boolean) properties this is OK. The top level properties are kept
```

Comment:
It would be good to add some more details here. Something like this:

The top-level properties of `OpenTelemetryLoggerOptions` are preserved to provide reflection access to the `ConfigurationBinder`. This is due to the fact that the first generic argument of `RegisterProviderOptions` in the runtime code is annotated with DAM(All).

Reply:
I changed the wording on this to hopefully make it clearer.